socorro/signature/rules.py

```diff
@@ -233,6 +232,63 @@ def normalize_frame(
         # Return module/module_offset
         return "{}@{}".format(module or "", strip_leading_zeros(module_offset))
 
+    def frame_generator(self, frames):
+        """Yields frames one at a time from a list of frames
+
+        Note: this treats inlines as individual frames and yields those when they're
+        available.
+
+        :arg frames: list of frame dicts
+
+        :returns: generator of frame data for normalization
+
+        """
+        for frame in frames:
+            inlines = frame.get("inlines") or []
+            for inline in inlines:
+                yield {
+                    "module": frame.get("module"),
+                    "function": inline.get("function"),
+                    "file": inline.get("file"),
+                    "line": inline.get("line"),
+                    "module_offset": frame.get("module_offset"),
+                    "offset": frame.get("offset"),
+                }
+
+            yield {
+                "module": frame.get("module"),
+                "function": frame.get("function"),
+                "file": frame.get("file"),
+                "line": frame.get("line"),
+                "module_offset": frame.get("module_offset"),
+                "offset": frame.get("offset"),
+            }
+
+    def create_frame_list(self, thread_data, make_modules_lower_case=False):
```

Comment:
`create_frame_list` only applies to c signatures, so I moved it to the `CSignatureTool`.